### PR TITLE
WIP - Homebrew automation

### PR DESCRIPTION
**Pull request by bobsummerwill**
This should be nearly all of the changes required to re-add Homebrew automation.

The exceptions are:
1. The hook-up in `release.sh` for the version number
2. Certificate so that the homebrew-ethereum Git commit works.
3. Uploading the `.tar.gz` to `builds.ethereum.org`.   Still haven't quite worked out how the metadata for the bottle themselves hooks up.

### CODE REVIEW

File: ethminer/PhoneHome.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ class PhoneHome : public jsonrpc::Client
             p.append(param1);
             p.append(param2);
             Json::Value result = this->CallMethod("report_benchmark",p);
-            if (result.isInt())
+            if (result.isIntegral())
```
Comment (chriseth) [inline ethminer/PhoneHome.h]: Why did you add this change?

**Comment by chriseth** [comment]
travis is failing, looks good otherwise.

File: ethminer/PhoneHome.h (hunk 1 above)
Comment (bobsummerwill) [inline ethminer/PhoneHome.h]: Ah - that was completely accidental commit.

Caused by the PhoneHome.h bug ->https://github.com/ethereum/cpp-ethereum/issues/3234

I will revert that back out, and see if I can work out where that file change is coming from within the build process.

Comment (bobsummerwill) [inline ethminer/PhoneHome.h]: See https://github.com/ethereum/cpp-ethereum/pull/3296.   Found the root cause.

**Comment by chfast** [comment]
Obsoleted.
